x/ecocredit/project.go

```diff
@@ -0,0 +1,34 @@
+package ecocredit
+
+import (
+	"fmt"
+	"regexp"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// reProjectID defines regular expression to check if the string has alphabets, numbers 
+// or combination of both characters and can be 2 ~ 16 characters long.
```

Comment:
i preferred the way you had it before, alphanumeric covers both cases

```suggestion
// reProjectID defines regular expression to check if the string contains only alphanumeric characters 
// and is between 2 ~ 16 characters long.
```